### DISCUSSION TITLE
How to set a date as tag in release-action?

### DISCUSSION THREAD

**Discussion by mwalter** [community/community#25766] [Actions]
<p>I would like to set a date as tag in the release-action. But how?<br>
This does not work:</p>
<pre><code class="lang-auto">    steps:
      - uses: ncipollo/release-action@v1
        with:
          artifacts: "my-artifact.zip"
          tag: ${{ env.NOW }}
          token: ${{ secrets.GITHUB_TOKEN }}
        env:
          NOW: $(date +'%Y-%m-%dT%H:%M:%S')
</code></pre>
<p>I get these errors:<br>
Errors:</p>
<ul>
<li>tag_name is not a valid tag</li>
<li>tag_name is not well-formed</li>
<li>Published releases must have a valid tag</li>
</ul>

**Comment by airtower-luna** [comment]
<p>You can’t use shell syntax in the <code>env:</code> definition. What you can do is use a <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" rel="nofollow noopener"><code>set-env</code> command</a> in a previous step:</p>
<pre><code class="lang-yaml">- name: set NOW
  run: |
    echo "::set-env name=NOW::$(date +'%Y-%m-%dT%H:%M:%S')"
</code></pre>
<p>After that <code>env.NOW</code> will be available. Note however that there’s a <a href="https://github.com/actions/runner/issues/528" rel="nofollow noopener">currently open bug</a> that may lead to the date being reformatted, a workaround has been discussed on this forum: <a href="https://github.community/t/iso-8601-string-in-job-output-is-automatically-parsed/117195">ISO 8601 string in job output is automatically parsed</a></p>

**Comment by mwalter** [comment]
<p>Did not work. Getting the same validation errors.<br>
How do I have to reference variable NOW in tag property?</p>
<pre><code class="lang-auto">${{ env.NOW }}?
${{ NOW }}?
$NOW?
${{ $NOW }}?
</code></pre>

**Comment by airtower-luna** [comment]
<p><code>${{ env.NOW }}</code> should work, yes. Mind that it the step using the variable must be a separate step after the one setting the environment variable. Saying more about possible reasons isn’t really possible without the workflow code, I’m afraid.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mwalter">@mwalter</a>,</p>
<p>The expression syntax <strong>${{ env.NOW }}</strong> is an correct way to access the custom env, just like <a class="mention" href="/airtower-luna">@airtower-luna</a> has suggested.</p>
<pre><code class="lang-yaml">steps:
  - name: set env
    run: echo "::set-env name=NOW::$(date +'%Y-%m-%dT%H:%M:%S')"

  - name: create release
    uses: ncipollo/release-action@v1
    with:
      artifacts: my-artifact
      tag: ${{ env.NOW }}
      token: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>This expression can correctly pass the value of <strong>env.NOW</strong> into the action “<strong>ncipollo/release-action@v1</strong>”.</p>
<p>The root cause of you still get the invalid <strong>tag_name</strong> error is that the format (<strong>DateTime</strong>) of the <strong>tag_name</strong> you set is not supported on GitHub.</p>
<pre><code class="lang-yaml">Errors:
- tag_name is not a valid tag
- tag_name is not well-formed
- Published releases must have a valid tag
</code></pre>
<p>I tried to manually create a release, and I used the same date-time format to set the tag (e.g. <strong>2020-06-22T02:11:39</strong>), then I got the “<strong>Invalid tag name</strong>” error.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8b0f58a6211801935ea8a52995fda7297097e3e7.png" data-download-href="/uploads/short-url/jQbflZpUz70gwJBZni1DMa7AnsP.png?dl=1" title="Invalid-tag-name" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181093170-3c647c63-0d11-428b-9da2-f1b8a930dd21.png" alt="Invalid-tag-name" data-base62-sha1="jQbflZpUz70gwJBZni1DMa7AnsP" width="690" height="318" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b0f58a6211801935ea8a52995fda7297097e3e7_2_690x318.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8b0f58a6211801935ea8a52995fda7297097e3e7_2_1035x477.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8b0f58a6211801935ea8a52995fda7297097e3e7.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b0f58a6211801935ea8a52995fda7297097e3e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Invalid-tag-name</span><span class="informations">1125×520 29.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
I recommend that you can reference to the “<strong>Tagging suggestions</strong>” to name the release tag.

**Comment by mwalter** [comment]
<p>Okay, thanks a lot! I just changed the pattern to:</p>
<pre><code class="lang-auto">echo "::set-env name=NOW::v$(date +'%Y%m%d.%H%M%S')"
</code></pre>
<p>This way the date is accepted by the release-action tag. It looks like that which satisfies my needs: v20200623.092227</p>

**Comment by ThioJoe** [comment]
<p>An update as of January 2022, when I tried to use the code mentioned above, I got a warning that GitHub deprecated the “set-env” command:</p>
<blockquote>
<p>The <code>set-env</code> command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the <code>ACTIONS_ALLOW_UNSECURE_COMMANDS</code> environment variable to <code>true</code>. For more information see: <a href="https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions: Deprecating set-env and add-path commands | GitHub Changelog</a></p>
</blockquote>
<p>Instead, this is what I got working (note I wanted my time in the format of “2021.12.31” ):</p>
<blockquote>
<p>run: echo “NOW=$(date +’%Y.%m.%d’)” &gt;&gt; $GITHUB_ENV</p>
</blockquote>
<p>and when using in a field somewhere, it’s the same as before, such as:</p>
<blockquote>
<p>title: “${{ env.NOW }}”</p>
</blockquote>
<p>So my in the context of all my steps it looks like this, where I wanted the tags and titles to be the date:</p>
<blockquote>
<pre><code>steps:
  # ...
  - name: set env
    run: echo "NOW=$(date +'%Y.%m.%d')" &gt;&gt; $GITHUB_ENV
    
  - name: "Build &amp; test"
    run: |
      echo "done!"

  - uses: "marvinpinto/action-automatic-releases@latest"
    with:
      repo_token: "${{ secrets.GITHUB_TOKEN }}"
      automatic_release_tag: "${{ env.NOW }}"
      prerelease: false
      title: "${{ env.NOW }}"
      files: |
        test.txt</code></pre>
</blockquote>
